tests/tests_geomstats/test_special_orthogonal.py

```diff
@@ -1,289 +1,1414 @@
-        result = self.so3.are_antipodals(rotation_mats1, rotation_mats2)
-        expected = gs.array([True, False])
+    def test_compose_and_inverse(self, point):
+        group = self.space(3, point_type="vector")
+        inv_point = group.inverse(point)
+        result = group.compose(point, inv_point)
+        expected = group.identity
+        self.assertAllClose(result, expected)
+        result = group.compose(inv_point, point)
         self.assertAllClose(result, expected)
 
-    def test_log_antipodals(self):
-        rotation_mat1 = gs.eye(3)
-        rotation_mat2 = gs.array([[1.0, 0.0, 0.0], [0.0, -1.0, 0.0], [0.0, 0.0, -1.0]])
-        with pytest.raises(ValueError):
-            self.so3.log(rotation_mat1, rotation_mat2)
-
-    def test_matrix_from_rotation_vector(self):
-        group = SpecialOrthogonal(n=3)
-        rot_vec_0 = gs.array([0.0, 0.0, 0.0])
-        result = group.matrix_from_rotation_vector(rot_vec_0)
-        expected = gs.eye(3)
+    def test_compose_regularize(self, point):
+        group = self.space(3, point_type="vector")
+        result = group.compose(point, group.identity)
+        expected = group.regularize(point)
         self.assertAllClose(result, expected)
 
-        rot_vec_1 = gs.array([gs.pi / 3.0, 0.0, 0.0])
-        result = group.matrix_from_rotation_vector(rot_vec_1)
-        expected = gs.array(
-            [
-                [1.0, 0.0, 0.0],
-                [0.0, 0.5, -gs.sqrt(3.0) / 2],
-                [0.0, gs.sqrt(3.0) / 2, 0.5],
-            ]
-        )
+        result = group.compose(group.identity, point)
+        expected = group.regularize(point)
         self.assertAllClose(result, expected)
 
-        rot_vec_3 = 1e-11 * gs.array([12.0, 1.0, -81.0])
-        angle = gs.linalg.norm(rot_vec_3)
-        skew_rot_vec_3 = 1e-11 * gs.array(
-            [[0.0, 81.0, 1.0], [-81.0, 0.0, -12.0], [-1.0, 12.0, 0.0]]
+    def test_compose_regularize_angles_close_to_pi(self, point):
+        group = self.space(3, point_type="vector")
+        result = group.compose(point, group.identity)
+        expected = group.regularize(point)
+        inv_expected = -expected
+        self.assertTrue(
+            gs.allclose(result, expected) or gs.allclose(result, inv_expected)
         )
-        coef_1 = gs.sin(angle) / angle
-        coef_2 = (1.0 - gs.cos(angle)) / (angle**2)
-        expected = (
-            gs.eye(3)
-            + coef_1 * skew_rot_vec_3
-            + coef_2 * gs.matmul(skew_rot_vec_3, skew_rot_vec_3)
+
+        result = group.compose(group.identity, point)
+        expected = group.regularize(point)
+        inv_expected = -expected
+        self.assertTrue(
+            gs.allclose(result, expected) or gs.allclose(result, inv_expected)
         )
-        result = group.matrix_from_rotation_vector(rot_vec_3)
+
+    @geomstats.tests.np_autograd_and_tf_only
+    def test_regularize_extreme_cases(self, elements_all):
```

Comment:
Shouldn't the loop for different inputs be different elements of smoke_data, instead of having assertAllClose inside the loop?